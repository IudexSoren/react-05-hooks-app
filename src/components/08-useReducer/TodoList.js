import TodoListItem from "./TodoListItem"


const TodoList = ({ todos, handleComplete, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {
        todos.map((todo, i) => {
          return (
            <TodoListItem
              key={ todo.id }
              {...todo}
              index={ i }
              handleComplete={ handleComplete }
              handleDelete={ handleDelete }
            />
          )
        })
      }
    </ul>
  )
}

export default TodoList
