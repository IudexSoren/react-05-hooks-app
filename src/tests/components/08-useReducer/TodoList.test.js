import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';
import { initialState } from '../../fixtures/demoTodos';

describe('Pruebas de TodoList', () => {

  const handleComplete = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(<TodoList
                            todos={ initialState }
                            handleComplete={ handleComplete }
                            handleDelete={ handleDelete }
                            />);

  test('Debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe renderizar los componentes TodoListItem correctamente', () => {
    expect(wrapper.find('TodoListItem').length).toBe(initialState.length);
    expect(wrapper.find('TodoListItem').at(0).prop('handleComplete')).toEqual(expect.any(Function));
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
  });

})
