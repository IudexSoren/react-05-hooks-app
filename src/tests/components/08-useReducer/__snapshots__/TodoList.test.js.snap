// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de TodoList Debe renderizar el componente correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    desc="Aprender React"
    done={false}
    handleComplete={[MockFunction]}
    handleDelete={[MockFunction]}
    id={1}
    index={0}
    key="1"
  />
  <TodoListItem
    desc="Aprender Node"
    done={false}
    handleComplete={[MockFunction]}
    handleDelete={[MockFunction]}
    id={2}
    index={1}
    key="2"
  />
</ul>
`;
